### ISSUE TITLE
Consider adding a `not { }` builder (or something similar) to make negating assertions easier

### ISSUE THREAD

**Issue by mkobit** [robfletcher/strikt#139]
This will reduce the size of the API surface and enable easier usage of composing existing assertions.

Possible example:

```kotlin
expectThat(listOf(1,2,3)).isNotEmpty()
```
To
```kotlin
expectThat(listOf(1,2,3)).not { isEmpty() }
```

Original discussion on Kotlin Slack - https://kotlinlang.slack.com/archives/CAR7KJ96J/p1543854162018400

**Comment by robfletcher** [comment]
There is a `not()` method that inverts downstream assertions but we could add a block one too.

**Comment by robfletcher** [comment]
Fixed in c26cb42267cbc12f621c25da70b93ca2bb909831

Available in v0.19.1
